xonsh/prompt/gitstatus.py

```diff
@@ -187,20 +187,24 @@ def gitstatus_prompt():
     if s.operations:
         ret += _get_def("OPERATION") + "|" + "|".join(s.operations)
     ret += "|"
-    if s.staged > 0:
+    if s.staged > 0 and _get_def("STAGED"):
```

Comment:
It would be nice if get_def was only called once for each of these

Reply:
That bothered me a little too. But how to do that nicely without assignment expressions?